tags/login.tag

```diff
@@ -31,43 +22,52 @@
 		}
 		logout() {
 			firebase.auth().signOut();
-      let a;
-      refUser.once("value").then(function(snapshot) {
-        a = snapshot.exists();
-      });
-      console.log(a);
 		}
 
     // Firebase authentication state listener
-		firebase.auth().onAuthStateChanged(userObj => {
-		  if (userObj) {
-		    this.user = userObj;
-				// start data listening
-        stopListening = database.collection('Users').onSnapshot(snapshot => {
-					snapshot.forEach(doc => {
-            let userInfo = {
-              id: firebase.auth().currentUser.uid,
-              name: firebase.auth().currentUser.displayName,
-              email: firebase.auth().currentUser.email
-            };
-             database.collection('Users').doc(firebase.auth().currentUser.uid).set(userInfo);
-             refUser = firebase.database().ref("Users/" + firebase.auth().currentUser.uid + '/Projects');
+    firebase.auth().onAuthStateChanged(userObj => {
+      if (userObj) {
+          this.user = userObj;
+            // start data listening
+            stopListening = database.collection('Users').onSnapshot(snapshot => {
+              snapshot.forEach(doc => {
+                let userInfo = {
+                  id: firebase.auth().currentUser.uid,
+                  name: firebase.auth().currentUser.displayName,
+                  email: firebase.auth().currentUser.email
+                };
+                database.doc('Users/' + firebase.auth().currentUser.uid).get()
+                  .then(docSnapshot => {
+                  if (!docSnapshot.exists) {
+                     database.collection('Users').doc(firebase.auth().currentUser.uid).set(userInfo);
+                   }
+                 });
+                 refUser = firebase.database().ref("Users/" + firebase.auth().currentUser.uid + '/Projects');
+            });
+            this.update();
           });
-					this.update();
-		    })
       }
       else {
-		    this.user = null;
-				stopListening();
-		  }
-			this.update();
-		});
-
-    //check if user has previous projects
-
-
-  // console.log(a);
-
+        this.user = null;
+        stopListening();
+      }
+      this.checkProject();
+      this.update();
+    });
+    // check if have Projects
+    checkProject(){
```

Comment:
this function exists in 2 different tags. is that on purpose?

Reply:
The function should be in this tag not in the create tag.